diffuzzer/src/fuzzing/objective/dash.rs

```diff
@@ -16,20 +16,23 @@ use crate::{
 
 use dash::{FileDiff, FileInfo, HasherOptions, get_diff};
 
-struct DashState {
+pub struct DashState {
+    pub(crate) fs_state: Vec<FileInfo>,
+    pub(crate) hash: u64,
+}
+
+struct DashProducer {
     fs_dir: RemotePath,
     fs_internal: RegexSet,
-    fs_state: Vec<FileInfo>,
-    hash: u64,
 }
 
-impl DashState {
-    pub fn update(
+impl DashProducer {
+    pub fn calculate(
         &mut self,
         cmdi: &dyn CommandInterface,
         dash_path: &RemotePath,
         output_path: &RemotePath,
-    ) -> anyhow::Result<()> {
+    ) -> DashState {
```

Comment:
Стараемся делать без паник
 ```suggestion
    ) -> anyhow::Result<DashState> {
```
Аналогично в Harness